docs/README.md

```diff
@@ -753,16 +753,151 @@ public static async handle(event: StockMoved, register: Register): Promise<void>
+```
+
+The generator will automatically create a file called `Product.ts` with a TypeScript class of the same name under the entities directory. You can still create the entity manually. Since the generator is not doing any magic, all you need is a class decorated as `@Command`. Anyway, we recommend you always to use the generator, because it handles the boilerplate code for you.
+
+Note:
+
+> Running the entity generator with an EntityName that already exists, it will override the content of the current one. In future releases, we will display a warning before overwriting anything. Meantime, if you missed a field, just add it to the class because in Booster, all the infrastructure and data structures are inferred from your code.
+
 #### The reducer function
 
+Booster generates the reducer function as a static method of the entity class. That function is called by the framework every time that an event of the
+specified type is emitted. **This function must be pure**, which means that no side effects should happen.
+
+The reducer has two parameters by default:
+
+* `event` - The event object that has triggered this reducer
+* `currentEntity?` - A possible previous entity generated from another call to a reducer. Note that **this parameter is optional**, and can be `undefined` if this reducer call is the first one, like when you just deployed your app.
+
+This is the place to write business logic for your data generation.
+
+Given this entity:
+
+```ts
+@Entity
+export class Cart {
+  public constructor(
+    public id: UUID,
+    readonly items: Array<CartItem>
+  ) {}
+
+  @Reduces(ProductAdded)
+  public static reduceProductAdded(
+    event: ProductAdded,
+    currentCart?: Cart
+  ): Cart {
+    const newItems = addToCart(event.item, currentCart)
+    return new Cart(event.cartID, newItems)
+  }
+
+  @Reduces(ProductRemoved)
+  public static reduceProductRemoved(
+    event: ProductRemoved,
+    currentCart?: Cart
+  ): Cart {
+    const newItems = removeFromCart(event.item, currentCart)
+    return new Cart(event.cartID, newItems)
+  }
+}
+```
+
+We can visualize reduction like this:
+
+![reducer process gif](img/reducer.gif)
+
 #### Aggregate data using entities
 
+You can create entities to aggregate data that would belong to different
+entities.
+
+As an example, let's say you have already an app that has a `Cart` entity and a `Profile` entity and you want to add a new feature to have a list of the users favorite items.
+
+Instead of polluting the `Profile` entity with the events of the `Cart`, you create a third entity: `ProfileFavorites`.
+
+Then create two event handlers:
+
+* One that's listening for `ProfileCreated` and emits a `FavoritesCreated` event
+* Another one that listens for `OrderPlaced` and emits a `NewItemsPurchased`
+
+Then, the reducers of the `ProfileFavorites` entity will be subscribed to
+these newly emitted events. Aggregating the required data into a single entity.
+
```

Comment:
I'm absolutely unsure about this section, would need help to rephrase it better

Reply:
I think we can reshape the example to make it clearer. I get the point but I have two concerns:

- I'm not sure the proposed approach is cleaner than polluting the Profile entity with Cart events. 
  Does it makes sense to react to `ProfileCreated` and `OrderPlaced` directly in `ProfileFavorites`?

- Why are the purchased items the favorite ones? 
  for me that relationship is not clear, wouldn't it be better to talk about a `PurchasesHistory` entity?

Reply:
Yep, we can make this easier to see with your second point.

Regarding the first one, I think Booster has a limitation: different entities cannot subscribe to the same events. But I have yet to confirm this, as I'm not sure.